src/node.cpp

```diff
@@ -114,7 +114,10 @@ Node::~Node()
     client->preadabort(this);
 
     // remove node's fingerprint from hash
-    client->mFingerprints.remove(this);
+    if (!client->mOptimizePurgeNodes)
+    {
+        client->mFingerprints.remove(this);
+    }
```

Comment:
If we don't remove the entry in `mFingerprints`, won't we be in risk of trying a remote copy for a fresh upload of a file with the same fingerprint (see `MegaClient::nodebyfingerprint()`). If so, the node has been deleted, so would need to check for side effects.

Reply:
:thinking: I've just checked, the `mOptimizePurgeNodes` is set only at `purgenodesusersabortsc()` and the `mFingerprints` is cleared right before deleting nodes, so it should be safe :smiley: 
However, I think a `mNodeCounters.clear()` is missing at the `purgenodesusersabortsc()` for consistency?

Reply:
yes good point, it's good practice to keep all the affected items together like that.   That one is cleared later on during the init() call, but I agree it's better to keep these together so I've added that call, thanks.